whole-tale/gwvolman
# enh: cache images based on r2d files checksum

Tag Tale images using the checksum of r2d files and r2d short_id. Then during `build_image` phase check if given image exists in the local registry. This allows for more efficient caching even for completely new tales.

### How to test?
1. Create a Tale. Run it. Shut it down.
1. Upload a file "hello.txt" (or anything not r2d related) to the workspace. Run the tale.
1. It should skip the build phase.
1. Upload `apt.txt` with any sane content.
1. Run the tale. It should build a new image.
1. Stop and delete the Tale.
1. Create a new Tale, upload the same `apt.txt`, run the tale.
1. It should skip the build phase.
